src/Grpc.Net.Client/Internal/Configuration/ConvertHelpers.cs

```diff
@@ -0,0 +1,106 @@
+        {
+            return statusCode switch
+            {
+                StatusCode.OK => "OK",
+                StatusCode.Cancelled => "CANCELLED",
+                StatusCode.Unknown => "UNKNOWN",
+                StatusCode.InvalidArgument => "INVALID_ARGUMENT",
+                StatusCode.DeadlineExceeded => "DEADLINE_EXCEEDED",
+                StatusCode.NotFound => "NOT_FOUND",
+                StatusCode.AlreadyExists => "ALREADY_EXISTS",
+                StatusCode.PermissionDenied => "PERMISSION_DENIED",
+                StatusCode.Unauthenticated => "UNAUTHENTICATED",
+                StatusCode.ResourceExhausted => "RESOURCE_EXHAUSTED",
+                StatusCode.FailedPrecondition => "FAILED_PRECONDITION",
+                StatusCode.Aborted => "ABORTED",
+                StatusCode.OutOfRange => "OUT_OF_RANGE",
+                StatusCode.Unimplemented => "UNIMPLEMENTED",
+                StatusCode.Internal => "INTERNAL",
+                StatusCode.Unavailable => "UNAVAILABLE",
+                StatusCode.DataLoss => "DATA_LOSS",
+                _ => throw new InvalidOperationException($"Unexpected status code: {statusCode}")
+            };
+        }
+
+        public static StatusCode ConvertStatusCode(string statusCode)
+        {
+            return statusCode.ToUpperInvariant() switch
+            {
+                "OK" => StatusCode.OK,
+                "CANCELLED" => StatusCode.Cancelled,
+                "UNKNOWN" => StatusCode.Unknown,
+                "INVALID_ARGUMENT" => StatusCode.InvalidArgument,
+                "DEADLINE_EXCEEDED" => StatusCode.DeadlineExceeded,
+                "NOT_FOUND" => StatusCode.NotFound,
+                "ALREADY_EXISTS" => StatusCode.AlreadyExists,
+                "PERMISSION_DENIED" => StatusCode.PermissionDenied,
+                "UNAUTHENTICATED" => StatusCode.Unauthenticated,
+                "RESOURCE_EXHAUSTED" => StatusCode.ResourceExhausted,
+                "FAILED_PRECONDITION" => StatusCode.FailedPrecondition,
+                "ABORTED" => StatusCode.Aborted,
+                "OUT_OF_RANGE" => StatusCode.OutOfRange,
+                "UNIMPLEMENTED" => StatusCode.Unimplemented,
+                "INTERNAL" => StatusCode.Internal,
+                "UNAVAILABLE" => StatusCode.Unavailable,
+                "DATA_LOSS" => StatusCode.DataLoss,
+                _ => int.TryParse(statusCode, out var number)
+                    ? (StatusCode)number
+                    : throw new InvalidOperationException($"Unexpected status code: {statusCode}")
+            };
+        }
+
+        public static TimeSpan? ConvertDurationText(string? text)
+        {
+            if (text == null)
+            {
+                return null;
+            }
+
+            if (text.Length > 0 && text[text.Length - 1] == 's')
+            {
+                return TimeSpan.FromSeconds(Convert.ToDouble(text.Substring(0, text.Length - 1), CultureInfo.InvariantCulture));
```

Comment:
I've change this to use decimal and then multiple the result to get ticks.

Since timers are only precise down to milliseconds, having the *exact* value shouldn't matter.